src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/ProjectConfigurationFileChangeDetector.cs

```diff
@@ -74,7 +74,10 @@ public async Task StartAsync(string workspaceDirectory, CancellationToken cancel
 
             // Start listening for file changes. If a project configuration file changes in any way we'll notify listeners.
 
-            _watcher = new FileSystemWatcher(workspaceDirectory, LanguageServerConstants.ProjectConfigurationFile)
+            // If we don't trim workspaceDirectory before passing it to FileSystemWatcher then when it eventually finds
+            // the file it reports the path as something like C:\some\dir\/deeper/dir/project.razor.json, while we'll be
+            // comparing it against C:\some\dir/deeper/dir/project.razor.json, and they won't match due to the extra \.
+            _watcher = new FileSystemWatcher(workspaceDirectory.TrimEnd('/','\\'), LanguageServerConstants.ProjectConfigurationFile)
```

Comment:
It is interesting that the FileSystemWatcher doesn't handle this correctly. It doesn't have any requirement about the trailing slash for the directory path. I'd imagine they would properly handle merging the directory path and path to the file that matches the filter.

Reply:
I believe the probelm is because we mix `/` and `\`, @NTaylorMullen was meantioning something simlar during standup, @NTaylorMullen is this the same thing/a similar thing? Are we about to stomp on each-other?

Reply:
My thing was different. But `Path.Combine` isn't great at combining differing path formats  so when we eventually normalize the path down into a single path format you get double `//`:

![image](https://user-images.githubusercontent.com/2008729/77960138-9e987180-728c-11ea-81d9-d08783bd2b7a.png)